pydantic/dataclasses.py

```diff
@@ -276,3 +276,15 @@ def rebuild_dataclass(
             raise_errors=raise_errors,
             types_namespace=types_namespace,
         )
+
+
+def is_pydantic_dataclass(_cls: type[Any]) -> TypeGuard[type[PydanticDataclass]]:
+    """Whether a class is a pydantic dataclass.
+
+    Args:
+        _cls: The class.
```

Comment:
I'm not sure what the point of naming the argument `_cls` is. Can we at least do `__cls` so that it's positional only?

Reply:
Looking at a few other function definitions on this module it seems normally `cls` or `_cls` is used.

Maybe we can change this to `cls` as it is on `rebuild_dataclass` ? https://github.com/pydantic/pydantic/blob/2fe4aff5a58502bfa312e41e36b52ebbf4a42b26/pydantic/dataclasses.py#L232-L233

Reply:
> Can we at least do __cls so that it's positional only?

`__cls` won't do any name mangling, so someone would still able to pass it as kwarg. I believe to make it positional only we would require py>=3.8 to use `/` syntax 🤔

So just want to double check if you still want `__cls`, I can make that change then. 

please review

Reply:
please review

Reply:
It doesn't do anything at runtime but it does matter semantically and to linters / type checkers: https://peps.python.org/pep-0484/#positional-only-arguments